src/core/Akka.Streams/Implementation/Fusing/Ops.cs

```diff
@@ -3838,9 +3839,29 @@ public Logic(SourceShape<T> shape, IAsyncEnumerable<T> enumerable) : base(shape)
             public override void PreStart()
             {
                 base.PreStart();
-                _completionCts = new CancellationTokenSource();
                 _enumerator = _enumerable.GetAsyncEnumerator(_completionCts.Token);
             }
+            
+            public override void PostStop()
+            {
+                try
+                {
+                    _completionCts.Cancel();
+                    _completionCts.Dispose();
+                }
+                catch
+                {
+                }
+                try
+                {
+                    _enumerator.DisposeAsync().ConfigureAwait(false);
```

Comment:
Just feels like the right thing to do.

Reply:
I think that this is the actual fix, the enumerator needs to be disposed for the resources to be released.

Reply:
Yep, me too